### PR TITLE
feat(rome_json_formatter): JSON Formatting object #2570

### PR DESCRIPTION
**Pull request by denbezrukov**
## Summary

Fix https://github.com/rome/tools/issues/2570 for Object value.

Open questions:
How can we reuse `separated` between json and js crate?

## Test Plan

`cargo test -p rome_json_formatter`

### CODE REVIEW

File: crates/rome_json_formatter/src/json/auxiliary/member.rs [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
 use rome_json_syntax::JsonMember;
-use rome_rowan::AstNode;
 
 #[derive(Debug, Clone, Default)]
 pub(crate) struct FormatJsonMember;
 
 impl FormatNodeRule<JsonMember> for FormatJsonMember {
     fn fmt_fields(&self, node: &JsonMember, f: &mut JsonFormatter) -> FormatResult<()> {
-        format_verbatim_node(node.syntax()).fmt(f)
+        let content = format_with(move |f| {
```
Comment (denbezrukov) [inline crates/rome_json_formatter/src/json/auxiliary/member.rs:10]: Prettier uses for json only one layout:

- https://github.com/prettier/prettier/blob/28d097bbf52377af5a87f37af1b2bc9d30b5e5ce/src/language-js/print/assignment.js#L131-L139
- https://github.com/prettier/prettier/blob/28d097bbf52377af5a87f37af1b2bc9d30b5e5ce/src/language-js/print/assignment.js#L43-L44

File: crates/rome_json_formatter/src/json/auxiliary/member.rs [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+                [
+                    group(&node.name().format()),
+                    node.colon_token().format(),
+                    space(),
+                    node.value().format()
+                ]
+            )
+        });
+
+        write!(f, [group(&content)])
```
[the remark this answers is not in the document]
Comment (MichaReiser) [reply crates/rome_json_formatter/src/json/auxiliary/member.rs:22]: Is it necessary to group the name? The name always is a string literal which should never break.

**Review by MichaReiser — APPROVED** [review]

File: crates/rome_formatter/src/utils.rs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+use rome_rowan::{Language, SyntaxNode};
+
+/// Returns true if this node contains newlines in trivias.
+pub fn node_has_leading_newline<L: Language>(node: &SyntaxNode<L>) -> bool {
```
[the remark this answers is not in the document]
Comment (denbezrukov) [reply crates/rome_formatter/src/utils.rs]: fixed

File: crates/rome_json_formatter/src/json/auxiliary/member.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (denbezrukov) [reply crates/rome_json_formatter/src/json/auxiliary/member.rs]: fixed

**Review by MichaReiser — APPROVED** [review]

**Comment by denbezrukov** [comment]
Hm, 
What happened with pipelines?Oo

**Pull request merged**